job/replay_worker_test.go

```diff
@@ -0,0 +1,160 @@
+package job_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/odpf/optimus/core/logger"
+
+	"github.com/google/uuid"
+	"github.com/odpf/optimus/job"
+	"github.com/odpf/optimus/mock"
+	"github.com/odpf/optimus/models"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReplayWorker(t *testing.T) {
+	dagStartTime, _ := time.Parse(job.ReplayDateFormat, "2020-04-05")
+	startDate, _ := time.Parse(job.ReplayDateFormat, "2020-08-22")
+	endDate, _ := time.Parse(job.ReplayDateFormat, "2020-08-26")
+	currUUID := uuid.Must(uuid.NewRandom())
+	dagRunStartTime := time.Date(2020, time.Month(8), 22, 2, 0, 0, 0, time.UTC)
+	dagRunEndTime := time.Date(2020, time.Month(8), 26, 2, 0, 0, 0, time.UTC)
+	jobSpec := models.JobSpec{
+		Name: "job-name",
+		Schedule: models.JobSpecSchedule{
+			StartDate: dagStartTime,
+			Interval:  "0 2 * * *",
+		},
+	}
+	replayRequest := &models.ReplayWorkerRequest{
+		ID:    currUUID,
+		Job:   jobSpec,
+		Start: startDate,
+		End:   endDate,
+		Project: models.ProjectSpec{
+			Name: "project-name",
+		},
+		DagSpecMap: map[string]models.JobSpec{
+			"job-name": jobSpec,
+		},
+	}
+	t.Run("Process", func(t *testing.T) {
+		t.Run("should throw an error when replayRepo throws an error", func(t *testing.T) {
+			ctx := context.Background()
+			replayRepository := new(mock.ReplayRepository)
+			defer replayRepository.AssertExpectations(t)
+			errMessage := "replay repo error"
+			replayRepository.On("UpdateStatus", currUUID, models.ReplayStatusInProgress, models.ReplayMessage{}).Return(errors.New(errMessage))
+
+			replaySpecRepoFac := new(mock.ReplaySpecRepoFactory)
+			defer replaySpecRepoFac.AssertExpectations(t)
+			replaySpecRepoFac.On("New", replayRequest.Job).Return(replayRepository)
+
+			worker := job.NewReplayWorker(replaySpecRepoFac, nil)
+			err := worker.Process(ctx, replayRequest)
+			assert.NotNil(t, err)
+			assert.Equal(t, errMessage, err.Error())
+		})
+		t.Run("should throw an error when scheduler throws an error", func(t *testing.T) {
+			logger.Init(logger.ERROR)
```

Comment:
We can do this at the top of the test file with InitWithWriter and write could be `ioutil.Discard`

Reply:
this doesn't look fixed?